e2e/provision/playbooks/roles/bootstrap/tasks/prep-gitea.yml

```
@@ -18,6 +18,30 @@
       metadata:
         name: "{{ gitea.k8s.namespace }}"
 
+- name: Create docker hub secret
+  when:
+    - lookup('ansible.builtin.env', 'DOCKERHUB_USERNAME') | length > 0
+    - lookup('ansible.builtin.env', 'DOCKERHUB_TOKEN') | length > 0
+  vars:
+    username: "{{ lookup('ansible.builtin.env', 'DOCKERHUB_USERNAME') }}"
+    token: "{{ lookup('ansible.builtin.env', 'DOCKERHUB_TOKEN') }}"
+  block:
+    - name: Create docker-registry secret
+      ansible.builtin.shell: kubectl create secret docker-registry dockerhub --docker-username={{ username }} --docker-password={{ token }} -n {{ gitea.k8s.namespace }}
```

Just one single suggestion to prevent issues with [`ansible-lint`](https://ansible.readthedocs.io/projects/lint/)

```console
command-instead-of-shell: Use shell only when shell functionality is required.
playbooks/roles/bootstrap/tasks/prep-gitea.yml:64 Task/Handler: Create docker-registry secret

no-changed-when: Commands should not change things if nothing needs doing.
playbooks/roles/bootstrap/tasks/prep-gitea.yml:64 Task/Handler: Create docker-registry secret
```

```suggestion
      ansible.builtin.command: >
        kubectl create secret docker-registry dockerhub --docker-username={{ username }}
        --docker-password={{ token }} -n {{ gitea.k8s.namespace }}
      changed_when: false
```